### PR TITLE
Cherry pick MinGW build fixes to stable-2.0

### PR DESCRIPTION
**Pull request by Biswa96**
This tries to fix some mingw build warnings as stated here https://github.com/FreeRDP/FreeRDP/issues/7067

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

File: include/freerdp/server/rdpdr.h [hunk 1]
```diff
@@ -35,20 +35,22 @@
...
 {
 	UINT32 NextEntryOffset;
 	UINT32 FileIndex;
-	UINT64 CreationTime;
-	UINT64 LastAccessTime;
-	UINT64 LastWriteTime;
-	UINT64 ChangeTime;
-	UINT64 EndOfFile;
-	UINT64 AllocationSize;
+	LARGE_INTEGER CreationTime;
```
Comment (akallabeth) [inline include/freerdp/server/rdpdr.h:43]: This change is problematic for stable branch as the API breaks with it.

**Review by akallabeth — CHANGES_REQUESTED** [review]
Overall LGTM, only the change in the public header (although correct) needs some discussion as it breaks public API

File: include/freerdp/server/rdpdr.h (hunk 1 above)
Comment (Biswa96) [reply include/freerdp/server/rdpdr.h:43]: According to MS docs [FILE_DIRECTORY_INFORMATION structure][1], those members are of LARGE_INTEGER type.

[1]: https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/ntifs/ns-ntifs-_file_directory_information

File: libfreerdp/core/nla.h [hunk 2]
```diff
@@ -58,7 +58,7 @@ FREERDP_LOCAL SEC_WINNT_AUTH_IDENTITY* nla_get_identity(rdpNla* nla);
 FREERDP_LOCAL NLA_STATE nla_get_state(rdpNla* nla);
 FREERDP_LOCAL BOOL nla_set_state(rdpNla* nla, NLA_STATE state);
 
-FREERDP_LOCAL BOOL nla_set_service_principal(rdpNla* nla, LPSTR principal);
+FREERDP_LOCAL BOOL nla_set_service_principal(rdpNla* nla, LPTSTR principal);
```
Comment (hardening) [inline libfreerdp/core/nla.h:61]: We avoid using `LPTSTR` that have different behaviours on msvc and gcc

**Pull request closed without merging**